src/array_partition.jl

```diff
@@ -256,26 +248,18 @@ end
 
 # workaround for https://github.com/SciML/RecursiveArrayTools.jl/issues/49
 function Base._unsafe_getindex(::IndexStyle, A::ArrayPartition,
-                               I::Vararg{Union{Real, AbstractArray}, N}) where {N}
+    I::Vararg{Union{Real, AbstractArray}, N}) where {N}
     # This is specifically not inlined to prevent excessive allocations in type unstable code
     shape = Base.index_shape(I...)
     dest = similar(A.x[1], shape)
     Base._unsafe_getindex!(dest, A, I...) # usually a generated function, don't allow it to impact inference result
     return dest
 end
 
-Base._maybe_reshape(::IndexCartesian, A::ArrayPartition, I::Vararg{Union{Real, AbstractArray}, N}) where {N} = Vector(A)
-
-"""
-    setindex!(A::ArrayPartition, v, i::Int, j...)
-
-Set the entry at index `j...` of the `i`th partition of `A` to `v`.
-"""
-Base.@propagate_inbounds function Base.setindex!(A::ArrayPartition, v, i::Int, j...)
-    @boundscheck 0 < i <= length(A.x) || throw(BoundsError(A.x, i))
-    @inbounds b = A.x[i]
-    @boundscheck checkbounds(b, j...)
-    @inbounds b[j...] = v
+function Base._maybe_reshape(::IndexCartesian,
```

Comment:
why was this removed?

Reply:
In 1.10 the internal implementation of `\` changed, and it indexes `ArrayPartition` with two indexes (the last of which is always `1`. That ends up calling this dispatch, and segfaults since the bounds checks are ignore (`@inbounds`). Adding methods for `Base.:\` and `ldiv!` ends up cascading into more methods for `mul!` and so on, which all need to only be defined for 1.10